### ISSUE TITLE
Getting System error: No module named 'TokenGeneratorByRange' when generating data

### ISSUE THREAD

**Issue by sugandhrise** [DiTEC-project/gnn-pressure-estimation#2]
Hi

I am running the code on Ubuntu and after fixing some hardcoded paths, I was able to execute python generator/EPYNET/ConfigCreator.py -r inputs/CTOWN.inp

`python generator/EPYNET/scenegenv7.py   --config gnn-pressure-estimation/config/v7.1/CTOWN_7v1__EPYNET_config.ini `

it gives `ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'` error!

Here is the full stack trace:
```
Start simulation...
saved_path =  datasets/CTOWN
skip nodes = ['R1']
#skip_nodes = 1
#skip_links = 0
exepected #result_nodes = 395 | Note that if attribute is 'demand', #results_nodes should be #junctions
exepected #result_links = 444
  0%|                                                                                                                                              | 0/20 [00:00<?, ?it/s]/home/sug/anaconda3/envs/gnn/lib/python3.9/site-packages/wntr/epanet/io.py:2076: UserWarning: Not all curves were used in "/home/sug/gnn-pressure-estimation/inputs/CTOWN.inp"; added with type None, units conversion left to user
  warnings.warn('Not all curves were used in "{}"; added with type None, units conversion left to user'.format(self.wn.name))
/home/sug/anaconda3/envs/gnn/lib/python3.9/site-packages/epynet/baseobject.py:63: UserWarning: requesting dynamic properties from an unsolved network
  warnings.warn("requesting dynamic properties from an unsolved network")
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 20/20 [00:02<00:00,  7.02it/s]
OK
2024-03-11 20:53:38,588	INFO worker.py:1553 -- Started a local Ray instance.
  0%|                                                                                                                                              | 0/20 [00:00<?, ?it/s]WARNING! Ray error The actor died because of an error raised in its creation task, ray::WDNRayExecutor.__init__() (pid=1341636, ip=192.168.1.82, repr=<Executorv7.FunctionActorManager._create_fake_actor_class.<locals>.TemporaryActor object at 0x7f47215d2550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'
traceback: Traceback (most recent call last):
ModuleNotFoundError: No module named 'TokenGeneratorByRange'
  5%|██████▋                                                                                                                               | 1/20 [00:00<00:06,  3.10it/s]WARNING! Ray error The actor died because of an error raised in its creation task, ray::WDNRayExecutor.__init__() (pid=1341636, ip=192.168.1.82, repr=<Executorv7.FunctionActorManager._create_fake_actor_class.<locals>.TemporaryActor object at 0x7f47215d2550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'
traceback: Traceback (most recent call last):
ModuleNotFoundError: No module named 'TokenGeneratorByRange'
WARNING! Ray error The actor died because of an error raised in its creation task, ray::WDNRayExecutor.__init__() (pid=1341636, ip=192.168.1.82, repr=<Executorv7.FunctionActorManager._create_fake_actor_class.<locals>.TemporaryActor object at 0x7f47215d2550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'
traceback: Traceback (most recent call last):
ModuleNotFoundError: No module named 'TokenGeneratorByRange'
WARNING! Ray error The actor died because of an error raised in its creation task, ray::WDNRayExecutor.__init__() (pid=1341636, ip=192.168.1.82, repr=<Executorv7.FunctionActorManager._create_fake_actor_class.<locals>.TemporaryActor object at 0x7f47215d2550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'
traceback: Traceback (most recent call last):
ModuleNotFoundError: No module named 'TokenGeneratorByRange'
WARNING! Ray error The actor died because of an error raised in its creation task, ray::WDNRayExecutor.__init__() (pid=1341636, ip=192.168.1.82, repr=<Executorv7.FunctionActorManager._create_fake_actor_class.<locals>.TemporaryActor object at 0x7f47215d2550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'
traceback: Traceback (most recent call last):
ModuleNotFoundError: No module named 'TokenGeneratorByRange'
WARNING! Ray error The actor died because of an error raised in its creation task, ray::WDNRayExecutor.__init__() (pid=1341636, ip=192.168.1.82, repr=<Executorv7.FunctionActorManager._create_fake_actor_class.<locals>.TemporaryActor object at 0x7f47215d2550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'
traceback: Traceback (most recent call last):
ModuleNotFoundError: No module named 'TokenGeneratorByRange'
WARNING! Ray error The actor died because of an error raised in its creation task, ray::WDNRayExecutor.__init__() (pid=1341636, ip=192.168.1.82, repr=<Executorv7.FunctionActorManager._create_fake_actor_class.<locals>.TemporaryActor object at 0x7f47215d2550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'
traceback: Traceback (most recent call last):
ModuleNotFoundError: No module named 'TokenGeneratorByRange'
WARNING! Ray error The actor died because of an error raised in its creation task, ray::WDNRayExecutor.__init__() (pid=1341636, ip=192.168.1.82, repr=<Executorv7.FunctionActorManager._create_fake_actor_class.<locals>.TemporaryActor object at 0x7f47215d2550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'
traceback: Traceback (most recent call last):
ModuleNotFoundError: No module named 'TokenGeneratorByRange'
WARNING! Ray error The actor died because of an error raised in its creation task, ray::WDNRayExecutor.__init__() (pid=1341636, ip=192.168.1.82, repr=<Executorv7.FunctionActorManager._create_fake_actor_class.<locals>.TemporaryActor object at 0x7f47215d2550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'
traceback: Traceback (most recent call last):
ModuleNotFoundError: No module named 'TokenGeneratorByRange'
WARNING! Ray error The actor died because of an error raised in its creation task, ray::WDNRayExecutor.__init__() (pid=1341636, ip=192.168.1.82, repr=<Executorv7.FunctionActorManager._create_fake_actor_class.<locals>.TemporaryActor object at 0x7f47215d2550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'
traceback: Traceback (most recent call last):
ModuleNotFoundError: No module named 'TokenGeneratorByRange'
WARNING! Ray error The actor died because of an error raised in its creation task, ray::WDNRayExecutor.__init__() (pid=1341636, ip=192.168.1.82, repr=<Executorv7.FunctionActorManager._create_fake_actor_class.<locals>.TemporaryActor object at 0x7f47215d2550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'
traceback: Traceback (most recent call last):
ModuleNotFoundError: No module named 'TokenGeneratorByRange'
WARNING! Ray error The actor died because of an error raised in its creation task, ray::WDNRayExecutor.__init__() (pid=1341636, ip=192.168.1.82, repr=<Executorv7.FunctionActorManager._create_fake_actor_class.<locals>.TemporaryActor object at 0x7f47215d2550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'
traceback: Traceback (most recent call last):
ModuleNotFoundError: No module named 'TokenGeneratorByRange'
WARNING! Ray error The actor died because of an error raised in its creation task, ray::WDNRayExecutor.__init__() (pid=1341636, ip=192.168.1.82, repr=<Executorv7.FunctionActorManager._create_fake_actor_class.<locals>.TemporaryActor object at 0x7f47215d2550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'
traceback: Traceback (most recent call last):
ModuleNotFoundError: No module named 'TokenGeneratorByRange'
WARNING! Ray error The actor died because of an error raised in its creation task, ray::WDNRayExecutor.__init__() (pid=1341636, ip=192.168.1.82, repr=<Executorv7.FunctionActorManager._create_fake_actor_class.<locals>.TemporaryActor object at 0x7f47215d2550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'
traceback: Traceback (most recent call last):
ModuleNotFoundError: No module named 'TokenGeneratorByRange'
WARNING! Ray error The actor died because of an error raised in its creation task, ray::WDNRayExecutor.__init__() (pid=1341636, ip=192.168.1.82, repr=<Executorv7.FunctionActorManager._create_fake_actor_class.<locals>.TemporaryActor object at 0x7f47215d2550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'
traceback: Traceback (most recent call last):
ModuleNotFoundError: No module named 'TokenGeneratorByRange'
WARNING! Ray error The actor died because of an error raised in its creation task, ray::WDNRayExecutor.__init__() (pid=1341636, ip=192.168.1.82, repr=<Executorv7.FunctionActorManager._create_fake_actor_class.<locals>.TemporaryActor object at 0x7f47215d2550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'
traceback: Traceback (most recent call last):
ModuleNotFoundError: No module named 'TokenGeneratorByRange'
WARNING! Ray error The actor died because of an error raised in its creation task, ray::WDNRayExecutor.__init__() (pid=1341636, ip=192.168.1.82, repr=<Executorv7.FunctionActorManager._create_fake_actor_class.<locals>.TemporaryActor object at 0x7f47215d2550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'
traceback: Traceback (most recent call last):
ModuleNotFoundError: No module named 'TokenGeneratorByRange'
WARNING! Ray error The actor died because of an error raised in its creation task, ray::WDNRayExecutor.__init__() (pid=1341636, ip=192.168.1.82, repr=<Executorv7.FunctionActorManager._create_fake_actor_class.<locals>.TemporaryActor object at 0x7f47215d2550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'
traceback: Traceback (most recent call last):
ModuleNotFoundError: No module named 'TokenGeneratorByRange'
WARNING! Ray error The actor died because of an error raised in its creation task, ray::WDNRayExecutor.__init__() (pid=1341636, ip=192.168.1.82, repr=<Executorv7.FunctionActorManager._create_fake_actor_class.<locals>.TemporaryActor object at 0x7f47215d2550>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'
traceback: Traceback (most recent call last):
ModuleNotFoundError: No module named 'TokenGeneratorByRange'
WARNING! Ray error The actor died because of an error raised in its creation task, ray::WDNRayExecutor.__init__() (pid=1341637, ip=192.168.1.82, repr=<Executorv7.FunctionActorManager._create_fake_actor_class.<locals>.TemporaryActor object at 0x7f2b6d7a55b0>)
  At least one of the input arguments for this task could not be computed:
ray.exceptions.RaySystemError: System error: No module named 'TokenGeneratorByRange'
traceback: Traceback (most recent call last):
ModuleNotFoundError: No module named 'TokenGeneratorByRange'
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 20/20 [00:00<00:00, 60.78it/s]

Simulation time: 5.050640344619751 seconds
Process run on 20 batches, total scenes: 100
Success/Expected: 0/10 scenes

```

Any help on what is causing the problem?

**Comment by zepengzhang** [comment]
Hi, have you solved this problem? I encounter the same problem.

**Comment by sugandhrise** [comment]
Yes, I had to tinker with the code a bit and was able to generate data but I am getting errors with train.py now.

**Comment by cuongth95** [comment]
Hello,
Thank you for your patience :) 
The bug is about module links. It has been fixed in commit cb9d06e6bb2c8e80f432a0accdfb01c3c067a95a.
